src/fetchers/gist-fetcher.js

```diff
@@ -0,0 +1,84 @@
+// @ts-check
+
+import axios from "axios";
+import { parseHTML } from "linkedom";
+
+/**
+ * @typedef {import('./types').GistData} GistData Gist data.
+ */
+
+/**
+ * Fetch GitHub gist information by given username and ID.
+ *
+ * @param {string} username Github username.
+ * @param {string} id Github gist ID.
+ * @returns {Promise<GistData>} Gist data.
+ */
+const fetchGist = async (username, id) => {
+  const response = (
+    await axios({
+      method: "get",
+      url: `https://api.github.com/gists/${id}`,
```

Comment:
I think we can get all information from the GraphQL API:

```graphql
{
  viewer {
    gist(name: "fb514518d708593b9a218eb61381e400") {
      name
      owner {
        login
      }
      stargazerCount
      forks {
        totalCount
      }
      files{
        language {
          name
        }
        size
      }
    }
  }
}
```

This way, we don't need to add the `parseHTML` library.

Reply:
Yes, you're right, this way is better since we send only one request instead of two and we don't need additional library for HTML parsing. I have updated the code.